### PR TITLE
Rubocop setup

### PR DESCRIPTION
**Pull request by jlestavel**
This PR is introducing the [Rubocop](https://rubocop.org/) Ruby linter/formatter for the `sidekiq-limit_fetch` repository.

To review this PR, I suggest that you use the "Hide whitespace option", as there are some classes that are re-indented because of the [`Style/ClassAndModuleChildren`](https://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Style/ClassAndModuleChildren) cop.

To generate this PR, I mostly use the Rubocop [autocorrect](https://docs.rubocop.org/rubocop/usage/autocorrect.html) option.

I came upon this subject while working on #142

### CODE REVIEW

File: sidekiq-limit_fetch.gemspec [hunk 1]
```diff
@@ -6,20 +9,24 @@ Gem::Specification.new do |gem|
...
+  gem.metadata['source_code_uri'] = 'https://github.com/deanpcmad/sidekiq-limit_fetch'
+  gem.metadata['changelog_uri'] = 'https://github.com/deanpcmad/sidekiq-limit_fetch/blob/master/CHANGELOG.md'
 
-  gem.files         = `git ls-files`.split($/)
-  gem.test_files    = gem.files.grep %r{^spec/}
-  gem.require_paths = %w(lib)
+  gem.files         = `git ls-files`.split($INPUT_RECORD_SEPARATOR)
+  gem.require_paths = %w[lib]
+
+  gem.required_ruby_version = '>= 2.7.0'
```
Comment (jlestavel) [inline sidekiq-limit_fetch.gemspec:22]: To respect the `Gemspec/RequiredRubyVersion` cop, I added this, according to:

https://github.com/deanpcmad/sidekiq-limit_fetch/blob/9bf29b248458c80593939f937c67c42ad4fae7e1/.github/workflows/ci.yml#L17

File: sidekiq-limit_fetch.gemspec [hunk 2]
```diff
@@ -6,20 +8,24 @@ Gem::Specification.new do |gem|
...
   gem.add_dependency 'redis', '>= 4.6.0'
-  gem.add_development_dependency 'redis-namespace', '~> 1.5', '>= 1.5.2'
+  gem.add_dependency 'sidekiq', '>= 4'
   gem.add_development_dependency 'appraisal'
-  gem.add_development_dependency 'rspec'
   gem.add_development_dependency 'rake'
+  gem.add_development_dependency 'redis-namespace', '~> 1.5', '>= 1.5.2'
+  gem.add_development_dependency 'rspec'
+  gem.add_development_dependency 'rubocop'
+  gem.metadata['rubygems_mfa_required'] = 'true'
```
Comment (jlestavel) [inline sidekiq-limit_fetch.gemspec:30]: I suppose that MFA is enabled? https://guides.rubygems.org/mfa-requirement-opt-in/

**Comment by jlestavel** [comment]
@deanpcmad rebased on `master`. Tests are failing but they fail on master too...

I think it would help future contributors to have Rubocop setup!

**Comment by deanpcmad** [comment]
Seeing as you've fixed the tests which are now in master, should be able to rebase :)

**Comment by jlestavel** [comment]
✅ that's done!

**Comment by deanpcmad** [comment]
Thanks! 👍

**Pull request merged**